pinot-broker/src/main/java/org/apache/pinot/broker/broker/ZkBasicAuthAccessControlFactory.java

```diff
@@ -79,11 +80,49 @@ public BasicAuthAccessControl(AccessControlUserCache userCache) {
 
         @Override
         public boolean hasAccess(RequesterIdentity requesterIdentity) {
-            return hasAccess(requesterIdentity, null);
+            return hasAccess(requesterIdentity, (BrokerRequest) null);
         }
 
         @Override
         public boolean hasAccess(RequesterIdentity requesterIdentity, BrokerRequest brokerRequest) {
+            Optional<ZkBasicAuthPrincipal> principalOpt = getPrincipalAuth(requesterIdentity);
+            if (!principalOpt.isPresent()) {
+                // no matching token? reject
+                return false;
+            }
+
+            ZkBasicAuthPrincipal principal = principalOpt.get();
+            if (brokerRequest == null || !brokerRequest.isSetQuerySource() || !brokerRequest.getQuerySource()
+                .isSetTableName()) {
+                // no table restrictions? accept
+                return true;
+            }
+
+            return principal.hasTable(brokerRequest.getQuerySource().getTableName());
+        }
```

Comment:
seems we can simply call the Set<String> tables API right?
```suggestion
        public boolean hasAccess(RequesterIdentity requesterIdentity, BrokerRequest brokerRequest) {
            if (brokerRequest == null || !brokerRequest.isSetQuerySource() || !brokerRequest.getQuerySource()
                .isSetTableName()) {
                // no table restrictions? accept
                return true;
            }
            return hasAccess(requestIdentity, Collections.singleton(brokerRequest.getQuerySource().getTableName())); 
        }
```